File: kotlin/hamming/src/main/kotlin/Hamming.kt
/**
 * Created by yannickgrenzinger on 16/08/2016.
 */
object Hamming {
    fun compute(a: String, b: String): Int {
        if (a.length != b.length) throw IllegalArgumentException()
        return a.zip(b).count { !it.first.equals(it.second) }
    }
}

File: clojure/hamming/src/hamming.clj
 (ns hamming)

 (defn ^:private compare-dna-by-strand
   [dna-1 dna-2]
   (map #(= (first %) (second %)) (map vector dna-1 dna-2)))

 (defn distance
   [dna-1 dna-2]
   (if (= (count dna-1) (count dna-2))
     (count (filter false? (compare-dna-by-strand dna-1 dna-2)))))
